src/python/bot/tasks/impact_task.py

```diff
@@ -96,11 +96,40 @@ def get_start_and_end_revision(regression_range, job_type):
   return start_revision, end_revision
 
 
+def get_component_impacts_from_url(component_name,
+                                   regression_range,
+                                   job_type,
+                                   platform=None):
+  """Gets component impact string using the build information url."""
+  start_revision, end_revision = get_start_and_end_revision(
+      regression_range, job_type)
+  if not end_revision:
+    return Impacts()
+
+  found_impacts = dict()
+  for build in ['stable', 'beta']:
+    build_revision_mappings = revisions.get_build_to_revision_mappings(platform)
```

Comment:
Just tested, the following block of code needs fixes to work. please see how mock tests can account for this, also please clean up current version and reupload.

```
  for build in ['stable', 'beta']:
    build_revision_mappings = revisions.get_build_to_revision_mappings(platform)
    if not build_revision_mappings:
      return Impacts()
    mapping = build_revision_mappings.get(build)
    if not mapping:
      return Impacts()
    chromium_revision = mapping['revision']
    component_revisions = revisions.get_component_revisions_dict(
        chromium_revision, 'default')
    if not component_revisions:
      return Impacts()
    result = False
    import six
    for key, value in six.iteritems(component_revisions):
      if value and 'name' in value and value['name'].lower() == component_name:
        branched_from = revisions.revision_to_branched_from(
            value['url'], value['rev'])
        if not branched_from:
          return Impacts()
        impact = get_impact({'revision': branched_from, 'version':mapping['version']}, start_revision, end_revision)
        found_impacts[build] = impact
        result =True
        break
    if not result:
      return Impacts()
```

Reply:
Thanks for the testing & suggestions, I'll go through it and see what I can write unit tests for.

Reply:
Done - please take another look. Thanks again!